src/beacon-api/app/routes/device_locations.py

```
@@ -0,0 +1,541 @@
+        "percentage": round(percentage, 2),
+        "optimal": 100 if percentage >= 90 else 0,
+        "good": 100 if 75 <= percentage < 90 else 0,
+        "fair": 100 if 50 <= percentage < 75 else 0,
+        "poor": 100 if percentage < 50 else 0
+    }
+
+
+@router.get("/devices/by-location", response_model=List[DeviceRead])
+async def get_devices_by_location(
+    *,
+    db: Session = Depends(get_db),
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)